Sources/Confidence/ConfidenceClient/NetworkTypeMapper.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import Common
+
+public enum NetworkTypeMapper {
+    public static func from(value: ConfidenceStruct) throws -> NetworkStruct {
+        NetworkStruct(fields: try value.compactMapValues(convertValue))
```

Comment:
q: is there a reason we use `compactMapValues` instead of map? do we expect null values in `convertValue`?

Reply:
now I see we can return nil from `convertValue`, do we want to skip those keys completely or send nil to the backend?

Reply:
Good point. Maybe we should pay more attention here. The "return nil" should never happen really, unless we have a bug, so I am tempted to just escalate the exception. We could catch it in the Engine, log the error locally, and delete the event from the batch (to avoid poison pills never leaving the system)

Reply:
Let's handle this on a next PR